feature-gate/program/src/instruction.rs

```diff
@@ -0,0 +1,126 @@
+//! Program instructions
+
+use solana_program::{
+    feature::Feature,
+    instruction::{AccountMeta, Instruction},
+    program_error::ProgramError,
+    pubkey::Pubkey,
+    rent::Rent,
+    system_instruction, system_program,
+};
+
+/// Feature Gate program instructions
+#[derive(Clone, Debug, PartialEq)]
+pub enum FeatureGateInstruction {
+    /// Submit a feature for activation.
+    ///
+    /// Note: This instruction expects the account to exist and be owned by the
+    /// system program. The account should also have enough rent-exempt lamports
+    /// to cover the cost of the account creation for a
+    /// `solana_program::feature::Feature` state prior to invoking this
+    /// instruction.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   0. `[w+s]` Feature account (must be a system account)
+    ///   1. `[]` System program
+    Activate,
```

Comment:
Since this doesn't add any new functionality, we could consider actually changing things up, maybe by having this take in a signer and some seeds to generate a PDA, which is the feature gate address! This way, it's possible to use a multisig or governance to activate a feature, instead of a normal keypair.

What do you think? We could also remove this instruction for now and just focus on the revoke, since that's the needed new function.

Reply:
Yep, cool idea for sure, and it's something we'll need to support. I've gone ahead and saved `Activate` for another PR, so we can continue discussion there!